analyzer.go

```diff
@@ -33,18 +32,15 @@ func (a *Analyzer) Analyze(image imgutil.Image, cache Cache) (platform.AnalyzedM
 		appMeta = platform.LayersMetadata{}
 	}
 
-	for _, bp := range a.Buildpacks {
-		if store := appMeta.MetadataForBuildpack(bp.ID).Store; store != nil {
-			if err := WriteTOML(filepath.Join(a.LayersDir, launch.EscapeID(bp.ID), "store.toml"), store); err != nil {
-				return platform.AnalyzedMetadata{}, err
-			}
+	if a.analyzeLayers() {
+		meta, err := a.LayerAnalyzer.RetrieveMetadataFrom(cache)
```

Comment:
```suggestion
		cacheMeta, err := a.LayerAnalyzer.RetrieveMetadataFrom(cache)
```

Reply:
Could `LayerAnalyzer.Analyze` return the cache metadata, and analyzer just ignores it? I feel like this would read better than having `RetrieveMetadataFrom` as a separate call, unless I'm missing something.